pkg/cmd/registry/artifact/crud/delete/delete.go

```diff
@@ -0,0 +1,187 @@
+				return runDelete(opts)
+			}
+
+			cfg, err := opts.Config.Load()
+			if err != nil {
+				return err
+			}
+
+			if !cfg.HasServiceRegistry() {
+				return errors.New("No service Registry selected. Use 'rhoas service-registry use' use to select your registry")
+			}
+
+			opts.registryID = fmt.Sprint(cfg.Services.ServiceRegistry.InstanceID)
+			return runDelete(opts)
+		},
+	}
+
+	cmd.Flags().BoolVarP(&opts.force, "yes", "y", false, "Delete without prompt")
+	cmd.Flags().StringVarP(&opts.outputFormat, "output", "o", "json", opts.localizer.MustLocalize("registry.cmd.flag.output.description"))
+
+	cmd.Flags().StringVarP(&opts.artifact, "artifact", "a", "", "Id of the artifact")
+	cmd.Flags().StringVarP(&opts.group, "group", "g", "", "Group of the artifact")
+	cmd.Flags().StringVarP(&opts.registryID, "instance-id", "", "", "Id of the registry to be used. By default uses currently selected registry")
+	flagutil.EnableOutputFlagCompletion(cmd)
+
+	return cmd
+}
+
+func runDelete(opts *Options) error {
+	logger, err := opts.Logger()
+	if err != nil {
+		return err
+	}
+
+	conn, err := opts.Connection(connection.DefaultConfigRequireMasAuth)
+	if err != nil {
+		return err
+	}
+
+	dataAPI, _, err := conn.API().ServiceRegistryInstance(opts.registryID)
+	if err != nil {
+		return err
+	}
+
+	if opts.group == "" {
+		logger.Info("Group was not specified. Using " + util.DefaultArtifactGroup + " artifacts group.")
+		opts.group = util.DefaultArtifactGroup
+	}
+
+	ctx := context.Background()
+	if opts.artifact == "" {
+		logger.Info("Artifact was not specified. Command will delete all artifacts in the group")
+		err = confirmDelete(opts, "Do you want to delete ALL ARTIFACTS from group "+opts.group)
+		if err != nil {
+			return err
+		}
+		request := dataAPI.ArtifactsApi.DeleteArtifactsInGroup(ctx, opts.group)
+		_, err = request.Execute()
+		if err != nil {
+			return registryinstanceerror.TransformError(err)
+		}
+		logger.Info("Artifacts in group " + opts.group + " deleted")
+	} else {
+		_, _, err := dataAPI.MetadataApi.GetArtifactMetaData(ctx, opts.group, opts.artifact).Execute()
+		if err != nil {
+			return errors.New("Artifact " + opts.artifact + " not found")
+		}
+		logger.Info("Deleting artifact " + opts.artifact)
+		err = confirmDelete(opts, "Do you want to delete artifact "+opts.artifact+" from group "+opts.group)
+		if err != nil {
+			return err
+		}
+		request := dataAPI.ArtifactsApi.DeleteArtifact(ctx, opts.group, opts.artifact)
+
+		_, err = request.Execute()
+		if err != nil {
+			return registryinstanceerror.TransformError(err)
+		}
+		logger.Info("Artifact deleted: " + opts.artifact)
+	}
+
+	return nil
+}
+
+func confirmDelete(opts *Options, message string) error {
+	if !opts.force {
+		var shouldContinue bool
+		confirm := &survey.Confirm{
+			Message: message,
+		}
+		err := survey.AskOne(confirm, &shouldContinue)
+		if err != nil {
+			return err
+		}
+
+		if !shouldContinue {
+			return errors.New("")
```

Comment:
Should this be an error with no message?

Reply:
Actually better to return nil

Reply:
```suggestion
			return errors.New("Command stopped")
```